sdk/eventhub/event-hubs/src/receiver.ts

```diff
@@ -301,49 +295,182 @@ export class EventHubConsumer {
+                this._context.connectionId,
+                this._baseConsumer && this._baseConsumer.name,
+                receivedEvents.length,
+                maxWaitTimeInSeconds
+              );
+              cleanUpBeforeReturn();
+              resolve(receivedEvents);
+            }
+          },
+          reject,
+          abortSignal
+        );
+
+        const addTimeout = (reuse: boolean = false): void => {
+          let msg = "[%s] Setting the wait timer for %d seconds for receiver '%s'.";
+          if (reuse) {
+            msg += " Receiver link already present, hence reusing it.";
+          }
+          log.batching(
+            msg,
+            this._context.connectionId,
+            maxWaitTimeInSeconds,
+            this._baseConsumer && this._baseConsumer.name
+          );
+
+          // resolve the operation's promise after the requested
+          // max number of seconds have passed.
+          timer = setTimeout(() => {
+            log.batching(
+              "[%s] Batching Receiver '%s', %d messages received when max wait time in seconds %d is over.",
+              this._context.connectionId,
+              this._baseConsumer && this._baseConsumer.name,
+              receivedEvents.length,
+              maxWaitTimeInSeconds
+            );
+            cleanUpBeforeReturn();
+            resolve(receivedEvents);
+          }, maxWaitTimeInSeconds * 1000);
+        };
+
+        if (!this._baseConsumer.isOpen()) {
+          try {
+            // initializing the baseConsumer will also add credits.
+            await this._baseConsumer.initialize();
+            // the operation may have been cancelled while the connection
+            // was being initialized. In this case, call abort.
+            if (abortSignal && abortSignal.aborted) {
+              return this._baseConsumer.abort();
+            }
+            addTimeout();
+          } catch (err) {
+            cleanUpBeforeReturn();
+            return reject(err);
+          }
+        } else {
+          addTimeout(true);
+        }
+
+        if (abortSignal) {
+          abortSignal.addEventListener("abort", onAbort);
+        }
+      });
+    };
+
+    const retryOptions = this._receiverOptions.retryOptions;
+    const jitterInSeconds = randomNumberFromInterval(1, 4);
+    const maxRetries =
+      retryOptions && retryOptions.maxRetries && retryOptions.maxRetries > 0
+        ? retryOptions.maxRetries
+        : Constants.defaultMaxRetries;
+    const delayInSeconds =
+      retryOptions && retryOptions.retryInterval && retryOptions.retryInterval > 0
+        ? retryOptions.retryInterval / 1000
+        : Constants.defaultDelayBetweenOperationRetriesInSeconds;
+
+    const config: RetryConfig<ReceivedEventData[]> = {
+      connectionHost: this._context.config.host,
+      connectionId: this._context.connectionId,
+      delayInSeconds: delayInSeconds + jitterInSeconds,
```

Comment:
Sure, though this does seem to make it inconsistent with what sender does. Since the receiver will still have credits available to receive events as long as the connection wasn't closed maybe that's ok though.

Reply:
We should remove jitter in sender as well. But thats for another day :)